assets/js/components/read-more/index.js

```diff
@@ -110,39 +130,53 @@ class ReadMore extends Component {
 	 * @param {obj} e event
 	 */
 	onClick( e ) {
-		const { isExpanded } = this.state;
-
 		e.preventDefault();
 
-		if ( isExpanded ) {
-			this.clampLines();
-		} else {
-			this.setState( {
-				content: this.originalContent,
-				isExpanded: ! isExpanded,
-			} );
-		}
+		const { isExpanded } = this.state;
+
+		this.setState( {
+			isExpanded: ! isExpanded,
+		} );
 	}
 
 	render() {
 		const { className } = this.props;
-		const { content } = this.state;
+		const { content, isClamped, isExpanded } = this.state;
 
 		if ( ! content ) {
 			return null;
 		}
 
+		if ( false === isClamped ) {
+			return (
+				<div className={ className }>
+					<div ref={ this.reviewContent }>
+						{ content }
+					</div>
+				</div>
+			);
+		}
+
 		return (
 			<div className={ className }>
 				<div
-					ref={ this.element }
-					dangerouslySetInnerHTML={ {
-						// `content` is the `review` parameter returned by the `reviews` endpoint.
-						// It's filtered with `wp_filter_post_kses()`, which removes dangerous HTML tags,
-						// so using it inside `dangerouslySetInnerHTML` is safe.
-						__html: content,
+					ref={ this.reviewSummary }
+					aria-hidden={ isExpanded }
+					style={ {
+						display: isExpanded && null !== isClamped ? 'none' : 'inherit',
+					} }
+				>
+					.
```

Comment:
What would you think about creating a state attribute named `clamppedContent` or something like that which might be initialized to `.` and is rendered here?

Then, in `clampLines()` we could do:
```diff
-this.reviewSummary.current.innerHTML = originalContent.slice( 0, markers.middle - 5 ) + ellipsis;
+this.setState( {
+	clamppedContent: originalContent.slice( 0, markers.middle - 5 ) + ellipsis,
+}
```
This way we don't need to touch the DOM with vanilla JS.

In a follow-up, we can even refactor this to avoid using the state and use [memoization](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization).

Reply:
@Aljullu Wouldn't that require a dangerous set, because innerHTML which we are working with here are not react components.

Reply:
Also prevents us getting the new height reliably since we cannot know if the state/render has yet completed.

Reply:
We can do both; set it during the loop so we can get the height, but store the final value in state and render with dangerouslySetHtml

Reply:
`componentDidUpdate` is needed it seems; otherwise re-render could occur mid-routine and break the logic.

Reply:
I don't think it's needed if you pass `maxHeight` when calling `this.clampLines()` in `componentDidMount()`.

In fact, I think `maxHeight` can be completely removed from the state. You only need to read it in `componentDidMount()` as you're currently doing and then pass it to `clampLines` as a parameter. Would that work?

Reply:
Yes thats fine. But the other parts (`componentDidUpdate`) are needed because height calculations are repeated as content is changed.

Reply:
No, wait. Sorry I confused myself. `clampLines` cannot be called from `componentDidMount`. As I said, there is no guarantee that the HTML has been rendered yet UNTIL `componentDidUpdate`. Otherwise you risk grabbing the height of the wrong content.

If having it in state bothers you it can be moved to `this.` again, but we cannot pass it around.